aten/src/ATen/native/cuda/Dropout.cu

```diff
@@ -117,12 +231,29 @@ fused_dropout_cuda(const Tensor& self, double p, Generator * gen_){
       self_info.collapseDims();
       ret_info.collapseDims();
       mask_info.collapseDims(); //ret and mask are collapsed to 1d contiguous tensor
-      switch (self_info.dims) {
-        case 1:
-            fused_dropout_kernel<scalar_t, accscalar_t, unsigned int, 1><<<grid, dim_block, 0, at::cuda::getCurrentCUDAStream()>>>(self_info, ret_info, mask_info, nelem, pa, rng_engine_inputs);
-            break;
-        default:
-            fused_dropout_kernel<scalar_t, accscalar_t, unsigned int, -1><<<grid, dim_block, 0, at::cuda::getCurrentCUDAStream()>>>(self_info, ret_info, mask_info, nelem, pa, rng_engine_inputs);
+
+      int vec_size = get_vector_size<scalar_t>(self, ret, mask);
+
+      if (vec_size > 1) {
+        switch (vec_size) {
+         case 8:
+          fused_dropout_kernel_vec<scalar_t, accscalar_t, unsigned int, 1, 8><<<grid, dim_block, 0, at::cuda::getCurrentCUDAStream()>>>(self_info, ret_info, mask_info, nelem, pa, rng_engine_inputs);
+          break;
+         case 4:
+          fused_dropout_kernel_vec<scalar_t, accscalar_t, unsigned int, 1, 4><<<grid, dim_block, 0, at::cuda::getCurrentCUDAStream()>>>(self_info, ret_info, mask_info, nelem, pa, rng_engine_inputs);
+          break;
+         case 2:
+          fused_dropout_kernel_vec<scalar_t, accscalar_t, unsigned int, 1, 2><<<grid, dim_block, 0, at::cuda::getCurrentCUDAStream()>>>(self_info, ret_info, mask_info, nelem, pa, rng_engine_inputs);
+          break;
+        }
+      } else {
+        switch (self_info.dims) {
+          case 1:
+              fused_dropout_kernel<scalar_t, accscalar_t, unsigned int, 1><<<grid, dim_block, 0, at::cuda::getCurrentCUDAStream()>>>(self_info, ret_info, mask_info, nelem, pa, rng_engine_inputs);
+              break;
+          default:
+              fused_dropout_kernel<scalar_t, accscalar_t, unsigned int, -1><<<grid, dim_block, 0, at::cuda::getCurrentCUDAStream()>>>(self_info, ret_info, mask_info, nelem, pa, rng_engine_inputs);
+        }
```

Comment:
Would it be more clear if you do
```C++
switch(vec_size) {
case 8:
   ....
.....
case 1:
   if (self_info.dims == 1) {
     ...
   } else {
     ...
   }
}
```

Reply:
I personally think it's much of a much-ness, so I have no real preference here. If you feel strongly about the different representation I can move to that

Reply:
No, it's not strong